addons/account/models/chart_template.py

```diff
@@ -1,991 +1,680 @@
-                       'anglo_saxon_accounting': self.use_anglo_saxon,
-                       'account_storno': self.use_storno_accounting,
-                       'bank_account_code_prefix': self.bank_account_code_prefix,
-                       'cash_account_code_prefix': self.cash_account_code_prefix,
-                       'transfer_account_code_prefix': self.transfer_account_code_prefix,
-                       'chart_template_id': self.id
-        })
-
-        #set the coa currency to active
-        self.currency_id.write({'active': True})
-
-        # When we install the CoA of first company, set the currency to price types and pricelists
-        if company.id == 1:
-            for reference in ['product.list_price', 'product.standard_price']:
-                try:
-                    tmp2 = self.env.ref(reference).write({'currency_id': self.currency_id.id})
-                except ValueError:
-                    pass
-
-        # Set the fiscal country before generating taxes in case the company does not have a country_id set yet
-        if self.country_id:
-            # If this CoA is made for only one country, set it as the fiscal country of the company.
-            company.account_fiscal_country_id = self.country_id
-        elif not company.account_fiscal_country_id:
-            company.account_fiscal_country_id = self.env.ref('base.us')
-
-        # Install all the templates objects and generate the real objects
-        acc_template_ref, taxes_ref = self._install_template(company, code_digits=self.code_digits)
-
-        # Set default cash discount write-off accounts
-        if not company.account_journal_early_pay_discount_loss_account_id:
-            company.account_journal_early_pay_discount_loss_account_id = self._create_cash_discount_loss_account(
-                company, self.code_digits)
-        if not company.account_journal_early_pay_discount_gain_account_id:
-            company.account_journal_early_pay_discount_gain_account_id = self._create_cash_discount_gain_account(
-                company, self.code_digits)
-
-        # Set default cash difference account on company
-        if not company.account_journal_suspense_account_id:
-            company.account_journal_suspense_account_id = self._create_liquidity_journal_suspense_account(company, self.code_digits)
-
-        if not company.account_journal_payment_debit_account_id:
-            company.account_journal_payment_debit_account_id = self.env['account.account'].create({
-                'name': _("Outstanding Receipts"),
-                'code': self.env['account.account']._search_new_account_code(company, self.code_digits, company.bank_account_code_prefix or ''),
-                'reconcile': True,
-                'account_type': 'asset_current',
-                'company_id': company.id,
-            })
+            company = self.env.company
+        if isinstance(company, int):
+            company = self.env['res.company'].browse([company])
```

Comment:
from the docstring, it's supposed to be already a browse record :point_up:

Reply:
Not anymore